res/res_pjsip_endpoint_identifier_ip.c

```diff
@@ -297,18 +326,54 @@ static struct ast_sip_endpoint *common_identify(ao2_callback_fn *identify_match_
 
 static struct ast_sip_endpoint *ip_identify(pjsip_rx_data *rdata)
 {
-	struct ast_sockaddr addr = { { 0, } };
+	struct ast_sockaddr_with_tp addr_with_tp = { { { 0, } }, };
+	pj_ansi_strncpy(addr_with_tp.tp, rdata->tp_info.transport->type_name, 8);
```

Comment:
why 8?

Reply:
It's the size of PJSIP param. https://docs.pjsip.org/en/latest/api/generated/pjsip/group/group__PJSIP__TRANSPORT.html#_CPPv4N15pjsip_transport9type_nameE
Replaced pj_ansi_strncpy to pj_ansi_strxcpy, and used sizeof()